pulp_python/app/tasks/publish.py

```diff
@@ -107,39 +107,27 @@ def write_simple_api(publication):
     )
     index_metadata.save()
 
-    def find_artifact():
-        _art = content_artifact.artifact
-        if not _art:
-            _art = models.RemoteArtifact.objects.filter(content_artifact=content_artifact).first()
-        return _art
+    if len(index_names) == 0:
+        return
 
-    for (name, canonical_name) in index_names:
-        project_dir = '{simple_dir}{name}/'.format(simple_dir=simple_dir, name=canonical_name)
-        os.mkdir(project_dir)
+    releases = (
+        python_models.PythonPackageContent.objects.filter(
+            pk__in=publication.repository_version.content
+        )
+        .values("name", "filename", "sha256")
+        .order_by("name")
+    )
 
-        packages = python_models.PythonPackageContent.objects.filter(name=name)
-        package_detail_data = []
-        for package in packages.iterator():
-            artifact_set = package.contentartifact_set.all()
-            for content_artifact in artifact_set:
-                artifact = find_artifact()
-                published_artifact = models.PublishedArtifact(
-                    relative_path=content_artifact.relative_path,
-                    publication=publication,
-                    content_artifact=content_artifact
-                )
-                published_artifact.save()
-
-                checksum = artifact.sha256
-                path = "../../{}".format(package.filename)
-                package_detail_data.append((package.filename, path, checksum))
-
-        metadata_relative_path = '{project_dir}index.html'.format(project_dir=project_dir)
+    def write_project_page():
+        name = index_names[ind][1]
+        project_dir = f'{simple_dir}{name}/'
+        os.mkdir(project_dir)
+        metadata_relative_path = f'{project_dir}index.html'
 
         with open(metadata_relative_path, 'w') as simple_metadata:
             context = Context({
                 'project_name': name,
-                'project_packages': package_detail_data
+                'project_packages': package_releases
```

Comment:
We should just be passing in the values we need explicitly as function arguments rather than capturing them from the external scope.  It's a little too "magic" and not really necessary (cool that it works, though).  Otherwise it's too easy for refactoring to accidentally break the code.